collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -186,6 +184,11 @@ object ImmutableArray extends StrictOptimizedSeqFactory[ImmutableArray] {
       case that: ofRef[_] => Arrays.equals(unsafeArray.asInstanceOf[Array[AnyRef]], that.unsafeArray.asInstanceOf[Array[AnyRef]])
       case _ => super.equals(that)
     }
+    override def slice(from: Int, until: Int): ImmutableArray[T] = {
+      val lo = scala.math.max(from, 0)
+      val hi = scala.math.min(until, length)
+      new ofRef(Arrays.copyOfRange[T](unsafeArray, lo, hi))
```

Comment:
Better!  But the current behavior (in 2.12) is to give an empty array, not throw an exception, when the range is of negative size.  So you either need to check `lo < hi` or take from `lo` to `math.max(lo, hi)`.

Reply:
thanks, fixed.